0xProject/0x-monorepo
# Remove generated wrappers from contract-wrappers

## Description

This PR removes generated code from `@0x/contract-wrappers`. Resolves #2058

We initially added generated code to `@0x/contract-wrappers` intending to remove `@0x/abi-gen-wrappers` completely. While attempting to replace dependencies on `@0x/abi-gen-wrappers` with `@0x/contract-wrappers` across all packages, I found a number of circular dependencies, mostly because of unit tests, coordinator wrapper, and some utility functions.

<details>

To deal with this, I started extracting components from `@0x/contract-wrappers` and created two new packages: a tests package and a package for the Coordinator wrapper.
https://github.com/0xProject/0x-monorepo/tree/multilang-2058/remove-abi-gen-wrappers

~If we pursue the direction of stripping `@0x/contract-wrappers` to the minimum where there's not much added to the generated code (thus requiring minimal dependencies), and putting everything we remove into separate modules, I think it's highly likely that we would eventually want to offer a a package that integrates these modules that are intended to be used together. E.g. generated wrappers + manually written coordinator wrapper. **This is what `@0x/contract-wrappers` already offers**, so we could just use it instead of stripping it down only to rewrite it later.~

~I propose the following mental model:~
~- `@0x/abi-gen-wrappers` - individual contract wrappers. low level libs are free to import only the wrappers they need, thus avoiding circular dependencies~
~- `@0x/contract-wrappers` - re-exports the generated wrappers, provides a convenience instantiation class, unit tests, coordinator wrapper, and other stuff that we want to add. We should be careful to only add code that adds new functionality, not just syntactic sugar, otherwise we end up in the same situation as before.~
</details>

For now, we will keep both packages and import the generated code where needed. After V3 release, we will be able to delete much of the handwritten code and can revisit extracting coordinator wrapper into its own package.

## Testing instructions

* Build the monorepo, unit tests should work
* Check that abi-gen-wrappers is updated

## Types of changes



 * Bug fix (non-breaking change which fixes an issue)




## Checklist:
